libbeat/publisher/queue/memqueue/ackloop.go

```diff
@@ -109,9 +114,6 @@ func (l *ackLoop) handleBatchSig() int {
 func (l *ackLoop) collectAcked() chanList {
 	lst := chanList{}
 
-	acks := l.lst.pop()
```

Comment:
Can you quickly comment for my understand why this is not needed anymore?

Reply:
The `pop` is already done in the for loop. But the for loop also checks the list entry has actually been ACKed. The `pop` operation removed did assume the entry has been ACKed, without checking if that's the case.

Reply:
Argh... the pop has been here, cause the entry has already been ACKed. The channel is stored in `l.sig` in the run loop.